projects/vdk-plugins/vdk-oracle/README.md

```diff
@@ -142,6 +142,63 @@ def run(job_input):
 
     job_input.send_object_for_ingestion(payload=payload, destination_table="test_table")
 ```
+
+#### Case Sensitivity
+
+**vdk-oracle supports only lower-case and upper-case payload keys.** Oracle is case-insensitive by default. This is a
+challenge when ingesting payloads and doing type and schema inference, so we've opted for the simplest solution to
+avoid confusion on the user side.
+
+**Valid Ingestion**
+
+```python
+def run(job_input):
+    payload = {
+        "id": "5",
+        "str_data": "string",
+        "int_data": "12",
+        "float_data": "1.2",
+        "bool_data": "False",
+        "timestamp_data": "2023-11-21T08:12:53",
+        "decimal_data": "0.1",
+    }
+
+    job_input.send_object_for_ingestion(payload=payload, destination_table="test_table")
+```
+
+```python
+def run(job_input):
+    payload = {
+        "ID": "5",
+        "STR_DATA": "string",
+        "INT_DATA": "12",
+        "FLOAT_DATA": "1.2",
+        "BOOL_DATA": "False",
+        "TIMESTAMP_DATA": "2023-11-21T08:12:53",
+        "DECIMAL_DATA": "0.1",
+    }
+
+    job_input.send_object_for_ingestion(payload=payload, destination_table="TEST_TABLE")
+```
+
+**Invalid ingestion**
+
+Will infer the schema, but won't insert correctly.
+
+```python
+def run(job_input):
```

Comment:
Does this throw an error? 
I think it needs to throw an error if we don'y support it.

Reply:
why it is that we can't just toLower everything?

Reply:
We currently toLower everything, but it's not ideal. Oracle is normally case insensitive, so people are used to uppercase identifier names, so we should support those as well.

However, if you put an identifier in quotes when running SQL, e.g. when you create tables, you make it case-sensitive and it will be treated this way. So if you do

```sql
create table test_table (
    id number,
    str_data varchar2(255),
    int_data number,
    "Float_Data" float,
    bool_data number(1),
    timestamp_data timestamp,
    "Decimal_Data" decimal(14,8),
    primary key(id))
```

And then `SELECT column_name, FROM user_tab_columns WHERE table_name = TEST_TABLE` (which is what we do when ingesting), you'll get

```sql
ID, STR_DATA, INT_DATA, Float_Data, BOOL_DATA, TIMESTAMP_DATA, Decimal_Data
```

Then the user might try to ingest a payload

```python
{
        "id": 5,
        "STR_DATA": "string",
        "int_data": 12,
        "FLOAT_DATA": 1.2,
        "bool_data": True,
        "TIMESTAMP_DATA": datetime.datetime.utcfromtimestamp(1700554373),
        "decimal_data": Decimal(0.1),
 }
```

We can select the columns and try to match a key in the dict. For `Float_Data`, we can check if `Float_Data`, `FLOAT_DATA` or `float_data` are in the payload, but that's pretty much it.